src/driver/nativescript/NativescriptDriver.ts

```diff
@@ -87,37 +85,43 @@ export class NativescriptDriver extends AbstractSqliteDriver {
 
     /**
      * Creates connection with the database.
+     *
+     * nativescript-sqlite driver constructor and execSQL()/all() methods return Promise
+     * Also as mentioned in nativescript-sqlite documentation:
+     *
+     * "You should choose either to use a promise or a callback;
+     *  you can use whichever you are most comfortable with
+     *  however, you CAN use both if you want;
+     *  but side effects WILL occur with some functions."
+     *
+     *  We choose Promise. So we should use Promises, when create connection and when make query
+     *  Let's use async/await
      */
-    protected createDatabaseConnection() {
-        return new Promise<void>((ok, fail) => {
-            const options = Object.assign({}, {
-                name: this.options.database,
-            }, this.options.extra || {});
-
-            new this.sqlite(options.name, (err: Error, db: any): any => {
-                if (err) return fail(err);
-
-                // use object mode to work with TypeORM
-                db.resultType(this.sqlite.RESULTSASOBJECT);
-
-
-                // we need to enable foreign keys in sqlite to make sure all foreign key related features
-                // working properly. this also makes onDelete work with sqlite.
-                db.execSQL(`PRAGMA foreign_keys = ON;`, [], (err: Error, result: any): any => {
-                    if (err) return fail(err);
-                    // We are all set
-                    ok(db);
-                });
-            });
-        });
+    protected async createDatabaseConnection() {
+        try {
+            const db = await new this.sqlite(this.options.database, this.options.extra);
+
+            // use object mode to work with TypeORM
+            db.resultType(this.sqlite.RESULTSASOBJECT);
+
+            // we need to enable foreign keys in sqlite to make sure all foreign key related features
+            // working properly. this also makes onDelete work with sqlite.
+            await db.execSQL(`PRAGMA foreign_keys = ON;`, []);
+
+            return db;
+        } catch (err) {
+            throw new Error(err);
+        }
     }
 
     /**
      * If driver dependency is not given explicitly, then try to load it via "require".
      */
     protected loadDependencies(): void {
-        this.sqlite = this.driver;
-        if (!this.driver) {
+        try {
+            this.sqlite = this.options.driver || require("nativescript-sqlite");
```

Comment:
this require line here will break webpack builds for people who want to use typeorm/browser, and do not have `nativescript-sqlite` in node modules (which, a react-native or web-sqlite user need not have).

Reply:
@championswimmer `react-native` driver have same line and it is not problem (i'm just get warning about that and nothing breaks).
https://github.com/typeorm/typeorm/blob/0a46810a026e147e6cac80d9a5ee16ea3b4bfe67/src/driver/react-native/ReactNativeDriver.ts#L92

Reply:
Yes but it was already added in v2.x
We cannot break builds without major version change. 

Keeping require('') lines does break react native builds, as people have reported back here.